plugins/woo-ai/src/shared/random-loading-message.tsx

```diff
@@ -0,0 +1,216 @@
+	__( '⏳ We appreciate your heroic patience.', 'woocommerce' ),
+];
+
+const veryLateLoadingPhrases = [
+	__(
+		'🎩 Please pardon our slowness; we’re fine-tuning our top hat tricks…',
+		'woocommerce'
+	),
+	__( '😅 Still here, not frozen… Honest!', 'woocommerce' ),
+	__( '🐢 I see the finish line! Hold on…', 'woocommerce' ),
+	__( '😂 Who knew AI could get stage fright?', 'woocommerce' ),
+	__( '🍿 Longer loading time? Popcorn, anyone?', 'woocommerce' ),
+	__( '🍵 Grab a cuppa, this might take a sec…', 'woocommerce' ),
+	__( '🐢 Quick as a…artificial turtle?', 'woocommerce' ),
+	__( '⏳ Time is an illusion, while waiting is an art…', 'woocommerce' ),
+	__( '😬 Might be going slower than dial-up…', 'woocommerce' ),
+	__( '⏳ Patience is a virtue, right?…', 'woocommerce' ),
+	__( '🤖 AI shame mode activated, optimizing faster…', 'woocommerce' ),
+	__( '💤 Should I play some Lo-Fi beats while we wait?', 'woocommerce' ),
+	__( '😅 You could bake a cake in this time…', 'woocommerce' ),
+	__( '🐢 Not typically this slow, I promise…', 'woocommerce' ),
+	__( '😬 So, how’s the weather?', 'woocommerce' ),
+	__(
+		"🏅 Congrats! You'll win gold in the waiting Olympics!",
+		'woocommerce'
+	),
+	__( '🐌 Hitching a ride with a snail…', 'woocommerce' ),
+	__( '🥶 Loading so slow, even glaciers are jealous…', 'woocommerce' ),
+	__( '🌳 Entertaining sloths with our speed…', 'woocommerce' ),
+	__( "♟️ Blame it on the AI's newfound love for chess.", 'woocommerce' ),
+	__( '🕰️ Feeling a bit like the nineties in here…', 'woocommerce' ),
+	__( '🐹 Maybe we need to upgrade our hamster wheel…', 'woocommerce' ),
+];
+
+// Function to return the shuffled phrases
+const getShuffledPhrases = () => {
+	return {
+		early: shuffleArray( earlyLoadingPhrases ),
+		late: shuffleArray( lateLoadingPhrases ),
+		veryLate: shuffleArray( veryLateLoadingPhrases ),
+	};
+};
+
+// Returns a random loading message from the list based on the elapsed time
+const getRandomLoadingPhrase = (
+	elapsedTime: number,
+	phrasesStack: { [ x: string ]: string[] }
+): string => {
+	let key = 'early';
+	if ( elapsedTime >= 6000 && elapsedTime < 16000 ) {
+		key = 'late';
+	} else if ( elapsedTime >= 16000 ) {
+		key = 'veryLate';
+	}
+
+	// Pop the first message from the stack and push it back in
+	const poppedMessage = phrasesStack[ key ].shift();
+
+	if ( ! poppedMessage ) {
+		return '';
+	}
+
+	phrasesStack[ key ].push( poppedMessage );
+
+	return poppedMessage;
+};
+
+const RandomLoadingMessage: React.FC< RandomLoadingMessageProps > = ( {
+	isLoading,
+} ) => {
+	const messageUpdateTimeout = useRef< number >();
+	const startTimeRef = useRef< number >( 0 );
+	const phrasesStack = getShuffledPhrases();
+	const [ currentMessage, setCurrentMessage ] = useState(
+		getRandomLoadingPhrase( 0, phrasesStack )
+	);
+
+	// Recursive function to update the message on an increasing time interval
+	const updateMessage = useCallback(
+		( delay: number ) => {
+			messageUpdateTimeout.current = window.setTimeout( () => {
```

Comment:
I think you should clear the timeout before setting it the event it didn't get cleared properly after already being run

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/38392/commits/bf0d72e85e0f25bb8414c04b5f115f8ef63a5a78